server/bootstrap/passport/strategies/bnet.js

```diff
@@ -10,29 +11,27 @@ module.exports = function(authBnetConfig){
         user
           .findByUserId(profile.id)
           .then(function(){
-            user.setValues({
-                oauthTokenEncrypted: user.encryptOauthToken(accessToken),
-                oauthType: "bnet"
-            }).updateTimeLatestLogin();
-            if (!user.exists()) {
-              user.setValues({
-                  userId: profile.id
-              });
-            }
-            user
-              .save()
-              .then(function(){
-                console.log('Save: success', user.getValues());
-                return done(null, profile);
-              })
-              .error(function(err){
-                console.log("BnetStrategy (2):", err);
-                return done(null, null);
-              });
+            log.debug("Updating existing user. 'profile':\n", profile);
           })
           .error(function(err){
-            console.log("BnetStrategy (1):", err);
-            return done(null, null);
+            log.debug("Creating new user. 'profile':\n", profile);
+          })
+          .finally(function(){
+            user.setValues({
+                oauthTokenEncrypted: user.encryptOauthToken(accessToken),
+                oauthType: "bnet",
+                userId: profile.id
```

Comment:
That's exactly why. `id` can be very ambiguous. Especially when - as you say - you have to join tables.

To avoid contextual complications, I vote for keeping the naming `userId` over `id`.

@Sikian : Got any input?